Fixed minor build issues and changed needle color to be vivid red like OEM cluster

AuRoN89 opened the pull request:
Added Node 21.11.1 as a MAX version because Node 22 is not supported. There is some kind of incompatibility with the used SvelteJS version
Forced Kivy==21.1.0 because github link is no longer valid and Kivy build fail due to that.
changed needle color to be vivid red like OEM cluster

AuRoN89 committed:
Changed ST / RS Theme needle color to bea solid red, like the oem cluster needles

AuRoN89 committed:
Forced Kivy wheel compile issues, by forcing version. Github link is no longer valid
Added KIVY_NO_ARGS = 1 to allows local testing

AuRoN89 committed:
Added a max node version due to incompatibilities (node 22 DOESN'T work)

AuRoN89 committed:
Fixed max Node version condition

craigkai commented:
I can take a look sometime today, I know the pinned Github version for Kivy was due to a bug with log files not being cleared when log level was set to production/error level. So that could be an issue if it wasn't fixed in the official version.

I can get @MattKai45 input on the style changes to the existing RS theme. The other changes seem reasonable on first look!

AuRoN89 committed:
Adding new backgrounds doesn't require .svelte changes anymore. Now the dropdown menu that list the images is dinamically generated based on the .png and the .jpg files located in the frontend Background folder.
Removed algnement.png from that folder to prevent it from being listed.

AuRoN89 commented:
I added a new commit to the PR.

Adding new backgrounds doesn't require .svelte changes anymore. Now the dropdown menu that list the images is dinamically generated based on the .png and the .jpg files located in the frontend Background folder.
Removed algnement.png from that folder to prevent it from being listed.

This change TRIGGER a warning in console logs due to svelte trying to import files from the public (static) folder. You can ignore it because it's atually importing just the path of the files. I think the only way to remove it is adding the npm svelte module that allows moving files during the build (and change the code accordingly), but it's really unecessary and you leave it as it is

AuRoN89 committed:
Added TCS Sport / Off Dynamic View conditions
Added DIST+, DIST-, RES+ and ACC buttons PID
Fixed Oil Temp and Coolant Temp Min / Max Values based on real canbus min / max values

AuRoN89 committed:
More Min / Max values fixes

AuRoN89 committed:
Changed ESC PID to be a single one

AuRoN89 commented:
I closed it and opened a new one in 2.0.0 branch with more changes.

Reject this one.
